src/cascadia/TerminalSettingsModel/ApplicationState.h

```diff
@@ -21,11 +22,21 @@ Module Name:
 // This macro generates all getters and setters for ApplicationState.
 // It provides X with the following arguments:
 //   (type, function name, JSON key, ...variadic construction arguments)
-#define MTSM_APPLICATION_STATE_FIELDS(X) \
-    X(std::unordered_set<winrt::guid>, GeneratedProfiles, "generatedProfiles")
+#define MTSM_APPLICATION_STATE_FIELDS(X)                                       \
+    X(std::unordered_set<winrt::guid>, GeneratedProfiles, "generatedProfiles") \
+    X(Windows::Foundation::Collections::IVector<winrt::Microsoft::Terminal::Settings::Model::WindowLayout>, PersistedWindowLayouts, "persistedWindowLayouts")
 
 namespace winrt::Microsoft::Terminal::Settings::Model::implementation
 {
+    struct WindowLayout : WindowLayoutT<WindowLayout>
+    {
+        WindowLayout(){};

```

Comment:
Does this compile without a constructor? Because we could maybe remove this...
Otherwise we can also write `WindowLayout() noexcept = default;`

Reply:
iirc I needed an empty constructor, but I can try again.

Reply:
When it comes to me you can feel free to pick out only the comments you agree with. 🙂